### PR TITLE
Improvement/payment methods apple pay

### PR DESCRIPTION
**Pull request by gcojocaru**
integrated apple pay button for payment methods screen

### CODE REVIEW

File: JudoKitObjCExampleApp/ViewController.m [hunk 1]
```diff
@@ -32,21 +32,21 @@
 #import "JudoKitObjC.h"
 
 typedef NS_ENUM(NSUInteger, TableViewContent) {
-    TableViewContentPayment,
```
Comment (zfoltin) [inline JudoKitObjCExampleApp/ViewController.m]: Let's try and stick with 4 spaces, shall we?

Comment (gcojocaru) [reply JudoKitObjCExampleApp/ViewController.m]: done

File: JudoKitObjCExampleApp/ViewController.m [hunk 2]
```diff
@@ -68,23 +68,23 @@ @implementation ViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-
+
```
Comment (zfoltin) [inline JudoKitObjCExampleApp/ViewController.m]: I think these have been removed in a previous PR

File: JudoKitObjCExampleApp/ViewController.m [hunk 3]
```diff
@@ -252,22 +252,41 @@ - (void)tableView:(nonnull UITableView *)tableView didSelectRowAtIndexPath:(nonn
 
 - (void)paymentMethodOption {
     JPAmount *amount = [[JPAmount alloc] initWithAmount:@"0.01" currency:self.currentCurrency];
-
+    NSArray *items = @[
+                       [[PaymentSummaryItem alloc] initWithLabel:@"Item 1"
+                                                          amount:[NSDecimalNumber decimalNumberWithString:@"0.01$"]],
```
Comment (zfoltin) [inline JudoKitObjCExampleApp/ViewController.m]: Let's remove the $ from these, it's ignored and a bit confusing

File: JudoKitObjCExampleApp/ViewController.m [hunk 4]
```diff
@@ -252,22 +252,41 @@ - (void)tableView:(nonnull UITableView *)tableView didSelectRowAtIndexPath:(nonn
...
                      consumerReference:self.reference
                         paymentMethods:PaymentMethodsAll
+               applePayConfiguratation: configuration
                            cardDetails:nil
                             completion:^(JPResponse * response, NSError * error) {
                                 if (error || response.items.count == 0) {
                                     if (error.domain == JudoErrorDomain && error.code == JudoErrorUserDidCancel) {
                                         [self dismissViewControllerAnimated:YES completion:nil];
-                                        return;
+                                        return; // BAIL
```
Comment (zfoltin) [inline JudoKitObjCExampleApp/ViewController.m]: Noooooo 😱 don't add these back please. Man, you really need to get on top of your git skills..

File: JudoKitObjCExampleApp/ViewController.m [hunk 5]
```diff
@@ -455,39 +474,26 @@ - (void)applePayPreAuthOperation {
...
+    
     ApplePayConfiguration *configuration = [self applePayConfigurationWithType:transactionType];
-
+    
     [self.judoKitSession invokeApplePayWithConfiguration:configuration
                                               completion:^(JPResponse *_Nullable response, NSError *_Nullable error) {
-
+                                                  
                                                   if (error || response.items.count == 0) {
-                                                      if (error.domain == JudoErrorDomain && error.code == JudoErrorUserDidCancel) {
```
Comment (zfoltin) [inline JudoKitObjCExampleApp/ViewController.m]: if error we do nothing?

File: JudoKitObjCExampleApp/ViewController.m [hunk 6]
```diff
@@ -510,30 +516,30 @@ - (NSString *)getSampleConsumerReference {
 }
 
 - (ApplePayConfiguration *)applePayConfigurationWithType: (TransactionType)transactionType {
-
+    
     NSArray *items = @[
                        [[PaymentSummaryItem alloc] initWithLabel:@"Item 1"
-                                                          amount:[NSDecimalNumber decimalNumberWithString:@"0.01"]],
+                                                          amount:[NSDecimalNumber decimalNumberWithString:@"0.01$"]],
```
Comment (zfoltin) [inline JudoKitObjCExampleApp/ViewController.m]: 🙅‍♂

File: JudoKitObjCExampleApp/ViewController.m [hunk 7]
```diff
@@ -252,22 +252,41 @@ - (void)tableView:(nonnull UITableView *)tableView didSelectRowAtIndexPath:(nonn
 
 - (void)paymentMethodOption {
     JPAmount *amount = [[JPAmount alloc] initWithAmount:@"0.01" currency:self.currentCurrency];
```
Comment (mpetrenco) [inline JudoKitObjCExampleApp/ViewController.m]: There is an 
`- (ApplePayConfiguration *)applePayConfigurationWithType: (TransactionType)transactionType`
 method that you can use to get the configuration, there's no need to create it again inside the payment method.

Also, shouldn't we handle both payment and preauth operations?

File: JudoKitObjCExampleApp/ViewController.m (hunk 2 above)
Comment (gcojocaru) [reply JudoKitObjCExampleApp/ViewController.m]: These have been added after running clang-format

**Pull request merged**